app/src/main/java/com/github/sdpcoachme/database/CachingStore.kt

```diff
@@ -245,59 +255,153 @@ class CachingStore(private val wrappedDatabase: Database,
     }
 
     /**
-     * Adds event to a user's schedule
+     * Gets the group event with the given id
+     */
+    fun getGroupEvent(groupEventId: String): CompletableFuture<GroupEvent> {
+        return wrappedDatabase.getGroupEvent(groupEventId)
+    }
+
+    /**
+     * Gets all group events
+     */
+    fun getAllGroupEvents(): CompletableFuture<List<GroupEvent>> {
+        return wrappedDatabase.getAllGroupEvents()
+    }
+
+    /**
+     * Gets all group events sorted by date
+     */
+    fun getAllGroupEventsByDate(): CompletableFuture<List<GroupEvent>> {
+        return getAllGroupEvents().thenApply { groupEvents ->
+            groupEvents.sortedBy { groupEvent ->
+                LocalDateTime.parse(groupEvent.event.start)
+            }
+        }
+    }
+
+    /**
+     * Gets all group events that have not yet occurred, sorted by date
+     */
+    fun getUpcomingGroupEventsByDate(): CompletableFuture<List<GroupEvent>> {
+        return getAllGroupEventsByDate().thenApply { groupEvents ->
+            groupEvents.filter { groupEvent ->
+                LocalDateTime.parse(groupEvent.event.start).isAfter(LocalDateTime.now())
```

Comment:
For better coherence, we can take out `LocalDateTime.now()` as a val above the `return`.